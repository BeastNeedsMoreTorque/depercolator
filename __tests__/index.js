import fs from 'async-file';
import resolveBin from 'resolve-bin';
import { execAsync } from 'async-child-process';
import path from 'path';

const COMMAND = path.resolve('./dist/index.js');

function mapExtension(source) {
  const ext = path.extname(source);
  return `${path.dirname(source)}/${path.basename(source, ext)}.jsx`;
}

async function run(filename, flags = '') {
  const source = path.resolve(filename);
  const outputPath = mapExtension(source);
  await execAsync(`${COMMAND} ${flags} ${source}`);

  const result = await fs.readFile(outputPath, 'utf8');
  fs.unlink(outputPath);
  return result;
}

describe('happy test', () => {
  it('converts the file', async () => {
    const result = await run('./fixtures/testfile.cjsx');
    expect(result).toMatchSnapshot();
  });
});

describe('invalid option', () => {
  it('throws the error', async () => {
    expect.assertions(1);
    await expect(run('./fixtures/badfile.cjsx')).rejects.toBeDefined();
  });
});

describe('decaffeinate options', () => {
  it('supports all of them', async () => {
    const decafPath = resolveBin.sync('decaffeinate');
    const { stdout } = await execAsync(`${decafPath} -h`);
    const flags = stdout.match(/\s-[\w-]+/g);
    expect(flags).toMatchSnapshot();
  });
});

describe('prettier options', () => {
  it('supports all of them', async () => {
    const prettierPath = resolveBin.sync('prettier');
    const { stdout } = await execAsync(`${prettierPath} -h`);
    const flags = stdout.match(/\s-[\w-]+/g);
    expect(flags).toMatchSnapshot();
  });
});
