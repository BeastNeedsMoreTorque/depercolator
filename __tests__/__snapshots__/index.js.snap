// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decaffeinate options supports all of them 1`] = `
Array [
  " -h",
  " --help",
  " --keep-commonjs",
  " --prefer-const",
  " --loose-default-params",
  " --loose-for-expressions",
  " --loose-for-of",
  " --loose-includes",
  " --allow-invalid-constructors",
  " --enable-babel-constructor-workaround",
]
`;

exports[`happy test converts the file 1`] = `
"import React from \\"react\\";

let testConstant = \\"foo\\";

export default React.createClass({
  render() {
    return (
      <div className=\\"hello\\">
        <div>
          Hello
        </div>
      </div>
    );
  }
});
"
`;

exports[`prettier options supports all of them 1`] = `
Array [
  " --write",
  " --list-different",
  " -l",
  " --stdin",
  " --print-width",
  " --tab-width",
  " --use-tabs",
  " --no-semi",
  " --single-quote",
  " --no-bracket-spacing",
  " --jsx-bracket-same-line",
  " --trailing-comma",
  " --parser",
  " --no-color",
  " --version",
  " -v",
]
`;
